### PR TITLE
Host external image assets locally

### PR DESCRIPTION
**Pull request by pierlon**
## Summary

Uses the built-in PHP server to host the required image assets when the `AMP_Image_Dimension_Extract_Download_Test` test class is run.

This should _finally_ fix the dreaded SVG HTTP error we've been battling with once and for all.

## Checklist

- [ ] My pull request is addressing an [open issue](https://github.com/ampproject/amp-wp/contributing/project-management.md#life-of-an-issue) (please create one otherwise).
- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/contributing/engineering.md#tests).
- [ ] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/contributing/engineering.md) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by pierlon** [comment]
Looking into why it failed :disappointed:.

File: tests/php/test-amp-image-dimension-extract-download.php [hunk 1]
```diff
@@ -5,21 +5,40 @@
...
+		parent::setUpBeforeClass();
+
+		/*
+		 * Start PHP built-in server to host images to test.
+		 * Not ideal to use remote URLs; mocking would be better for performance, but FasterImage doesn't provide means to do this.
+		 */
+		$host      = self::get_host();
+		$host_root = self::get_host_root();
+
+		exec( "php -S {$host} -t {$host_root} >/dev/null 2>/dev/null &" ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.system_calls_exec
```
[the remark this answers is not in the document]
Comment (pierlon) [reply tests/php/test-amp-image-dimension-extract-download.php]: Resolved in https://github.com/ampproject/amp-wp/pull/4366/commits/fd5a5516083a5ce19100d034e7d11d65e908a8d3.

[the remark this answers is not in the document]
Comment (pierlon) [reply tests/php/test-amp-image-dimension-extract-download.php]: It was killed whenever the test suite is done, but now its killed when the class is being torn down:

https://github.com/ampproject/amp-wp/blob/fd5a5516083a5ce19100d034e7d11d65e908a8d3/tests/php/test-amp-image-dimension-extract-download.php#L67-L71

File: tests/php/test-amp-image-dimension-extract-download.php [hunk 2]
```diff
@@ -12,31 +12,90 @@
...
+		$url = wp_parse_url( self::get_server_url() );
 
-		/*
-		 * Start PHP built-in server to host images to test.
-		 * Not ideal to use remote URLs; mocking would be better for performance, but FasterImage doesn't provide means to do this.
-		 */
-		$host      = self::get_host();
-		$host_root = self::get_host_root();
+		if ( ! isset( $url['host'], $url['port'] ) ) {
+			throw new \Error( 'A host and port needs to be set to start the PHP server' );
```
Comment (westonruter) [inline tests/php/test-amp-image-dimension-extract-download.php]: What is `Error`? Do you mean `Exception`?

```suggestion
			throw new \Exception( 'A host and port needs to be set to start the PHP server' );
```

File: tests/php/test-amp-image-dimension-extract-download.php [hunk 3]
```diff
@@ -12,31 +12,90 @@
...
+			if ( ! $response instanceof WP_Error ) {
+				$started = true;
+				break;
+			}
+
+			sleep( 1 );
+		}
+
+		if ( ! $started ) {
+			throw new \Error( 'Failed to start the PHP server' );
```
Comment (westonruter) [inline tests/php/test-amp-image-dimension-extract-download.php]: See above.

```suggestion
			throw new \Exception( 'Failed to start the PHP server' );
```

File: tests/php/test-amp-image-dimension-extract-download.php (hunk 2 above)
Comment (pierlon) [reply tests/php/test-amp-image-dimension-extract-download.php]: Yes :man_facepalming:.

**Pull request merged**